image-encapsulate.cpp

```diff
@@ -2,10 +2,10 @@
 #include "brightener.h"
 
 int main() {
-	auto image = std::make_unique<Image>(512, 512);
+	auto image = std::make_shared<Image>(512, 512);
 	std::cout << "Brightening a 512 x 512 image\n";
-	ImageBrightener brightener(std::move(image));
-	if (brightener.ValidateImage()) {
+	ImageBrightener brightener(image);
+	if (image->pixels != nullptr) {
```

Comment:
Can optimize a bit further - if image is not valid (no pixels), then there's no need to instantiate an `ImageBrightener`